jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/ProcessDefinitionDataSetProvider.java

```diff
@@ -0,0 +1,133 @@
+import org.dashbuilder.dataset.def.DataColumnDef;
+import org.dashbuilder.dataset.def.DataSetDef;
+import org.dashbuilder.dataset.filter.CoreFunctionFilter;
+import org.dashbuilder.dataset.filter.DataSetFilter;
+import org.dashbuilder.dataset.impl.DataSetMetadataImpl;
+import org.dashbuilder.dataset.sort.DataSetSort;
+import org.dashbuilder.dataset.sort.SortOrder;
+import org.jbpm.workbench.ks.integration.ConsoleDataSetLookup;
+import org.jbpm.workbench.pr.model.ProcessDefinitionDataSetDef;
+import org.jbpm.workbench.pr.model.ProcessDefinitionDataSetProviderType;
+import org.jbpm.workbench.pr.model.ProcessSummary;
+import org.jbpm.workbench.pr.service.ProcessRuntimeDataService;
+
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.COL_ID_PROCESSNAME;
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.PROCESS_DEFINITION_DATASET;
+
+@ApplicationScoped
+public class ProcessDefinitionDataSetProvider extends BeanDataSetProvider  {
+
+    @Inject
+    protected ProcessRuntimeDataService processRuntimeDataService;
+
+    @Override
+    public DataSetProviderType getType() {
+        return new ProcessDefinitionDataSetProviderType();
+    }
+
+    @Override
+    public DataSetMetadata getDataSetMetadata(DataSetDef def) throws Exception {
+
+        List<String> columnNames = new ArrayList<>();
+        List<ColumnType> columnTypes = new ArrayList<>();
+
+        List<DataColumnDef> columns = def.getColumns();
+        for (DataColumnDef column : columns) {
+            columnNames.add(column.getId());
+            columnTypes.add(column.getColumnType());
+        }
+
+        return new DataSetMetadataImpl(def, def.getUUID(), -1, def.getColumns().size(), columnNames, columnTypes, -1);
+    }
+
+    @Override
+    public DataSet lookupDataSet(DataSetDef def, DataSetLookup lookup) throws Exception {
+
+        ProcessDefinitionDataSetDef processDefinitionDataSetDef = (ProcessDefinitionDataSetDef) def;
+        final DataSetBuilder dsBuilder = DataSetFactory.newDataSetBuilder();
+        for (final DataColumnDef columnDef : processDefinitionDataSetDef.getColumns()) {
+            dsBuilder.column(columnDef.getId(), columnDef.getColumnType());
+        }
+
+        ConsoleDataSetLookup consoleDataSetLookup = (ConsoleDataSetLookup) lookup;
+        processDefinitionDataSetDef.setServerTemplateId(consoleDataSetLookup.getServerTemplateId());
+
+        String defaultColumnSort = COL_ID_PROCESSNAME;
+        boolean defaultSortOrder = false;
+
+        DataSetSort dataSetSort = lookup.getFirstSortOp();
+        if (dataSetSort != null && dataSetSort.getColumnSortList().size() > 0) {
+            defaultColumnSort = dataSetSort.getColumnSortList().get(0).getColumnId();
+            defaultSortOrder = dataSetSort.getColumnSortList().get(0).getOrder().equals(SortOrder.ASCENDING);
+        }
+
+        final AtomicReference<String> searchTextReference = new AtomicReference<String>();
```

Comment:
what is the need for using an AtomicReference in here and not a simple String type?

Reply:
lambda need final variable，if use the final String, I can not reassign it new value.

Reply:
but it doesnt like a good use of AtomicReference, perhaps try todo a separated method to get the text String you need, something like:
```suggestion
        protected String getText(DataSetLookup lookup) {
        for (DataSetFilter filter : lookup.getOperationList(DataSetFilter.class)) {
            for(ColumnFilter columnFilter : filter.getColumnFilterList()){
                if (columnFilter instanceof CoreFunctionFilter) {
                    CoreFunctionFilter coreFunctionFilter = (CoreFunctionFilter) columnFilter;
                    if (coreFunctionFilter.getParameters().size() > 0) {
                        return coreFunctionFilter.getParameters().get(0).toString();
                    }
                }
            }
        }
        return "";
    }
```

Reply:
@bxf12315 could you address this please, thanks

Reply:
if I use the refactor method, I got unexpected return value
error: incompatible types
https://stackoverflow.com/questions/53480923/error-incompatible-types-unexpected-return-value-java-8

Reply:
Can you propose an alternative that does not use an AtomicReference, as this API is intended for concurrency uses cases.